apis/v1alpha1/cluster_defaults.go

```diff
@@ -47,11 +47,11 @@ func SetClusterSpecDefaults(cs *CrdbClusterSpec) {
 	}
 
 	if cs.Cache == "" {
-		cs.Cache = "25%"
+		cs.Cache = "$(expr $MEMORY_LIMIT_MIB / 4)MiB"
 	}
 
 	if cs.MaxSQLMemory == "" {
-		cs.MaxSQLMemory = "25%"
+		cs.MaxSQLMemory = "$(expr $MEMORY_LIMIT_MIB / 4)MiB"
```

Comment:
@chrisseto this is not working, I think, because we do not run an exec command. We cannot run an exec command because it does not work on openshift. Any ideas?

Reply:
we are actually running an exec in the container - the entrypoint script does that (see [cockroach.sh](https://github.com/cockroachdb/cockroach/blob/master/build/deploy/cockroach.sh)).  We can't explicitly do it outside of a bash script is all.

Reply:
Here are a couple of other options to try: `((MAX_SQL=MIB_LIMIT/4))` or `let MAX_SQL=$MIB_LIMIT/4`

Reply:
As resources are just being set via the CRD, couldn't we safely do the math in the operator? Otherwise we'd have to wrap everything in `/bin/sh` to get the envvars to evaluate.

Reply:
What would we do in the case where a customer didn't set a resource request or limit?

Reply:
`((MAX_SQL=MIB_LIMIT/4))` won't work.  I am not a shell script expert ... LOL.  Maybe we run the match before we run the command and set it there?

Reply:
that's worth a try @chrislovecnm

Reply:
> What would we do in the case where a customer didn't set a resource request or limit?

Either default back to `25%` or not set it at all? It might be better to not set it and then allow users to set the values manually with extra flags?